packages/next/build/webpack-config.ts

```diff
@@ -118,7 +118,9 @@ function errorIfEnvConflicted(config: NextConfigComplete, key: string) {
 
 function isResourceInPackages(resource: string, packageNames?: string[]) {
   return packageNames?.some((p: string) =>
-    new RegExp('[/\\\\]node_modules[/\\\\]' + p + '[/\\\\]').test(resource)
+    resource.includes(
+      path.sep + pathJoin('node_modules', p.replace(/\//g, path.sep)) + path.sep
```

Comment:
Will this work for the case where `pnpm` symlinks the package to a different name that includes the version e.g. `.pnpm/react@18.2.0...`

Reply:
IIRC Webpack has an option `resolve.symlink` and is enabled by default.

Reply:
Yeah I need to verify pnpm and Yarn PnP. `isResourceInPackages` itself accepts the resource location that is already resolved by webpack (which uses enhanced-resolve under the hood). So here we just need to ensure that it can correctly detect the package name / subpath.

Reply:
@ijjk Just confirmed that `pnpm` works because it always has the `"/node_modules/<package>"` substring in the resolved path: `/Users/shu/Documents/git/next.js/node_modules/.pnpm/@swc+helpers@0.4.11/node_modules/@swc/helpers/lib/_extends.js`